lib/iris/_lazy_data.py

```diff
@@ -135,3 +135,64 @@ def multidim_lazy_stack(stack):
         result = da.stack([multidim_lazy_stack(subarray)
                            for subarray in stack])
     return result
+
+
+def convert_nans_array(array, nans=None, result_dtype=None):
+    """
+    Convert a :class:`~numpy.ndarray` that may contain one or more NaN values
+    to either a :class:`~numpy.ma.core.MaskedArray` or a
+    :class:`~numpy.ndarray` with the NaN values filled.
+
+    Args:
+
+    * array:
+        The :class:`~numpy.ndarray` to be converted.
+
+    Kwargs:
+
+    * nans:
+        If `nans` is None, then raise an exception if the `array` contains
+        any NaN values (default).
```

Comment:
My take on it is, what are my NaN going to be replace with
- `nans=fill_value` is replace NaNs with the `fill_value`
- `nans=ma.masked` is replace my NaNs with a mask
- `nan=None` is I don't care, there shouldn't be any NaNs in my data, barf if there is

I'm open to a group take on this, as I've been staring at it way too long ... but I'm kinda wedded to the short and sweet `nans` name ... thoughts?